sdks/python/apache_beam/dataframe/frames_test.py

```diff
@@ -651,6 +651,18 @@ def test_series_getitem(self):
     s.index = s.index.map(float)
     self._run_test(lambda s: s[1.5:6], s)
 
+  def test_series_truncate(self):
+    s = pd.Series(['a', 'b', 'c', 'd', 'e', 'f'])
+    self._run_test(lambda s: s.truncate(before=1, after=3), s)
+
+  def test_dataframe_truncate(self):
+    df = pd.DataFrame({
+        'A': list('abcde'), 'B': list('fghij'), 'C': list('klmno')
+    },
+                      index=[1, 2, 3, 4, 5])
+    self._run_test(lambda df: df.truncate(before=1, after=3), df)
+    self._run_test(lambda df: df.truncate(before='A', after='B', axis=1), df)
```

Comment:
I think there's a subtle bug in the axis=1 case: if the columns aren't already in sorted order, your implementation will sort them. If you change this test like so, it should reveal that bug:

```suggestion
    df = pd.DataFrame({
        'C': list('abcde'), 'B': list('fghij'), 'A': list('klmno')
    },
                      index=[1, 2, 3, 4, 5])
    self._run_test(lambda df: df.truncate(before=1, after=3), df)
    self._run_test(lambda df: df.truncate(before='A', after='B', axis=1), df)
```

I think the best solution is to create a different expression for axis=1 vs axis=0, and only do the `sort_index()` in the axis=0 case.